### PR TITLE
EVG-14268 Add ParentPatchId field to Task struct

### PR DESCRIPTION
**Pull request by malikchaya2**

### CODE REVIEW

File: rest/model/task.go [hunk 1]
```diff
@@ -66,6 +66,7 @@ type APITask struct {
 	TaskGroupMaxHosts       int                 `json:"task_group_max_hosts,omitempty"`
 	Blocked                 bool                `json:"blocked"`
 	Requester               *string             `json:"requester"`
+	ParentPatchID           *string             `json:"parent_patch_id"`
```
Comment (ybrill) [inline rest/model/task.go:69]: I was thinking we'd just set the Version field

File: model/task/task.go [hunk 2]
```diff
@@ -133,6 +133,9 @@ type Task struct {
 	// patch request
 	Requester string `bson:"r" json:"r"`
 
+	// tasks that are part of a child patch will store the id of the parent patch
+	ParentPatchID string `bson:"parent_patch_id" json:"parent_patch_id,omitempty"`
```
Comment (ybrill) [inline model/task/task.go:137]: should it be `omitempty` in bson too?

File: model/build/build.go [hunk 3]
```diff
@@ -65,6 +65,9 @@ type Build struct {
 	// patch request
 	Requester string `bson:"r" json:"r,omitempty"`
 
+	// builds that are part of a child patch will store the id of the parent patch
+	ParentPatchID string `bson:"parent_patch_id" json:"parent_patch_id,omitempty"`
```
Comment (ybrill) [inline model/build/build.go:69]: should it be `omitempty` in bson too?

File: rest/model/task.go (hunk 1 above)
Comment (malikchaya2) [reply rest/model/task.go:69]: oooh good idea, then we don't need to do it in APITask!

**Review by ybrill — APPROVED** [review]

**Comment by malikchaya2** [comment]
evergreen merge

**Comment by malikchaya2** [comment]
evergreen merge

**Pull request merged**